src/equations/equations.jl

```diff
@@ -42,6 +42,22 @@ Common choices of the `conversion_function` are [`cons2cons`](@ref) and
 """
 function varnames end
 
+"""
+    get_variable_index(variable, equations, conversion_function = cons2cons)
+
+Return the index of `variable` in `varnames(conversion_function, equations)` if available.
+Otherwise, return an error.
+"""
+@inline function get_variable_index(variable, equations,
+                                    conversion_function = cons2cons)
+    index = findfirst(==(variable), varnames(conversion_function, equations))
+    if isnothing(index)
+        error("$variable is no valid variable.")
+    end
+
+    return index
+end
```

Comment:
```suggestion
# Return the index of `varname` in `varnames(solution_variables, equations)` if available.
# Otherwise, return an error.
"""
@inline function get_variable_index(varname, equations;
                                    solution_variables = cons2cons)
    index = findfirst(==(varname), varnames(solution_variables, equations))
    if isnothing(index)
        error("$variable is no valid variable.")
    end

    return index
end
```

I suggest a slight modification to make the conversion function a kwarg and name it `solution_variables` in accordance with our other uses of the conversion functions.

Also, I changed your docstring into a comment. Since we tentatively consider all docstring-annotated functions as part of our public API, it makes it hard to change it later (since it would be a breaking change).

@ranocha it would be nice to get a quick look from you as well on this

Reply:
Is this performance critical? This is something I would need to know how to proceed further when looking at the following aspects.
- `@inline` required?
- Explicit specialization on `solution_variables` required?
- Throwing an appropriate exception and not a generic `error`?

Reply:
Thanks @sloede.

> Is this performance critical?

No, it is not. I call this function only twice at the beginning of the simulation.

> Explicit specialization on `solution_variables` required?

Right now, it's not. I just need the index of conservative variables. I don't need this feature for the MCl as well. Do you want me to remove this functionality?

> Throwing an appropriate exception and not a generic `error`?

How would you do this?

Reply:
>  > Explicit specialization on solution_variables required?
> 
> Right now, it's not. I just need the index of conservative variables. I don't need this feature for the MCL as well. Do you want me to remove this functionality?

It is possible that the implementation of MCL for general equations requires some kind of this functionality. This is because different variables are limited there. For Euler, it is just `v1`, `v2` and `E`, which can be done by adding `rho_`. But for more complex equations, a `solution_variable` function like `cons2seq` with `seq` for sequential probably would be nice.
But as I said, this is not implemented yet.

Reply:
Since it's not performance critical, please remove the explicit `@inline` annotation.

> > Throwing an appropriate exception and not a generic `error`?
> 
> How would you do this?


`throw(ArgumentError("bla"))`

Reply:
Thanks for the input. I implemented it in [89f1b6e](https://github.com/trixi-framework/Trixi.jl/pull/1648/commits/89f1b6eaa200a869b84bb822fbedd751e97f92e5).